lib/services/local/userpreferences.go

```diff
@@ -177,6 +178,14 @@ func overwriteValues(dst, src protoreflect.ProtoMessage) error {
 func overwriteValuesRecursive(dst, src protoreflect.Message) {
 	src.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
 		switch {
+		case fd.IsMap():
+			srcMap := v.Map()
+			dstMap := dst.Mutable(fd).Map()
+
+			srcMap.Range(func(k protoreflect.MapKey, vv protoreflect.Value) bool {
+				dstMap.Set(k, vv)
+				return true
+			})
```

Comment:
just curious why this and the function is required?

Reply:
I was running into an issue when updating user preferences because its a "map" type and not a "message" type. So I had to first check if the incoming value to the method was a map in order to update it differently